core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1060,139 +1047,6 @@ public void callOptionsExecutor() {
         TimeUnit.SECONDS.toNanos(ManagedChannelImpl.SUBCHANNEL_SHUTDOWN_DELAY_SECONDS));
   }
 
-  @Test
-  public void nameResolutionFailed() {
```

Comment:
These tests don't seem to be added anywhere else

Reply:
I was trying to make sure these cases are covered in DnsNameResolverTest

- I updated the existing test resolve_addressFailure_neverLookUpServiceConfig(), where the name resolution lookup itself fails, to check that a retry task is scheduled.
- I added resolve_emptyResult_notAccepted() to cover the situation nameResolverReturnsEmptySubLists_resolutionRetry() here deals with (resolution works, but the LB rejects the results because the result has no addresses)
- nameResolverReturnsEmptySubLists_optionallyAllowed() is covered by resolve_emptyResult() (here the result is accepted, even if there are no addresses)
- nameResolutionFailed_delayedTransportShutdownCancelsBackoff() is special and no longer valid I believe as ManagedChannelImpl no longer controls the life cycle of the retry task and there's no way for DnsNameResolver to have visibility into transports and their delayed statuses.

Reply:
Thanks